src/implementation/aarch64/neon.rs

```diff
@@ -85,7 +85,19 @@ impl SimdU8Value {
     #[inline]
     #[allow(clippy::cast_ptr_alignment)]
     unsafe fn load_from(ptr: *const u8) -> Self {
-        Self::from(vld1q_u8(ptr))
+        // WORKAROUND:
+        // The vld1q_u8 intrinsic is currently broken, it treats it as individual
+        // byte loads so the compiler sometimes decides it is a better to load
+        // individual bytes to "optimize" a subsequent SIMD shuffle
+        //
+        // This code forces a full 128-bit load.
+        let mut dst = core::mem::MaybeUninit::<uint8x16_t>::uninit().assume_init();
```

Comment:
Here you are calling `assume_init` on an uninitialised piece of memory, which is undefined behaviour IIRC. You should call `assume_init` after the copy.

Reply:
Fixed. I shouldn't have blindly taken that from the stdarch sources of SSE2, see [here](https://github.com/rust-lang/stdarch/blob/6495bb0e33578443c21764655a4dd8b55399c008/crates/core_arch/src/x86/sse2.rs#L1193) and [here](https://github.com/rust-lang/stdarch/blob/6495bb0e33578443c21764655a4dd8b55399c008/crates/core_arch/src/x86/sse2.rs#L2762).